learning-progression.md

```diff
@@ -164,22 +172,32 @@ if (false) {
 
 ### Step 6: Data types
 
+#### 1. Study
+
+The basic functions of a computer are to store and process electrical signals which represent binary numbers, 1's and 0's. These numbers are used to represent all of the information in our computers, from a simple text document to an HD movie.
+
+To start thinking about the most basic type of data we can store, it is helpful to use the analogy of a light switch. When the switch is set to the higher power state it is on, which can be interpretted to mean true or 1. When the switch is flipped to the lower power state it is off, false, or 0. In this analogy, we can see that the same piece of data (the light switch being on or off) can be used to represent different kinds of information. 
+
+As such, one of the most rudimentary types of data in javascript is a "Boolean," which can has two possible states: "true" and "false." We can apply this in the example that we used in the previous step to allow us to more easily switch which block of code in the "if-else" statement is run.
+
+To declare a variable with a type (such as boolean) in javascript, the following template can be used: "let _name_ : _type_". The keyword "let" indicates that the line that follows will be to declare a variable. The "_name_" can be replaced with any name that is valid for the variable. And the "_type_" indicates how the stored data will be interpretted. For example, if we want to say that we have a variable called "isHeart," which will store a boolean, we would enter "let isHeart : boolean". With this code, we have declared our variable and given it a type, however it is not holding value. Since it has been declared as a boolean, we can either assign it a value of "true" or "false." To assign this variable a value or change the value it currently holds, we could enter "isHeart = true" or "isHeart = false".
+
+Additionally, javascript allows us to declare our variable and assign it an initial value in one line by typing "let _name_ : _type_ = _value_". In our example, to declare "isHeart" as a boolean and assign it an initial value "true," we would type "let isHeart : boolean = true".
```

Comment:
Anchor with the same example from above, showing the two statements fused together.
```javascript
let myVar : boolean = true
```